plugins/outputs/kinesis/kinesis.go

```diff
@@ -243,8 +244,7 @@ func (k *KinesisOutput) Write(metrics []telegraf.Metric) error {
 
 		r = append(r, &d)
 
-		if sz == 500 {
-			// Max Messages Per PutRecordRequest is 500
+		if sz >= k.maxRecordsPerRequest {
```

Comment:
Switched to `>=` to be more defensive.  Should only ever evaluate to `true` on `==`, but if there was a bug, the alternative is worse.